core/src/main/java/greencity/controller/UserController.java

```diff
@@ -388,15 +405,20 @@ public ResponseEntity<Boolean> checkIfTheUserIsOnline(
      */
     @Operation(summary = "Get user profile statistics by id")
     @ApiResponses(value = {
-        @ApiResponse(responseCode = "200", description = HttpStatuses.OK),
-        @ApiResponse(responseCode = "400", description = HttpStatuses.BAD_REQUEST),
-        @ApiResponse(responseCode = "401", description = HttpStatuses.UNAUTHORIZED),
+            @ApiResponse(responseCode = "200", description = HttpStatuses.OK),
+            @ApiResponse(responseCode = "400", description = HttpStatuses.BAD_REQUEST),
+            @ApiResponse(responseCode = "401", description = HttpStatuses.UNAUTHORIZED),
+            @ApiResponse(responseCode = "403", description = HttpStatuses.FORBIDDEN)
     })
     @GetMapping("/{userId}/profileStatistics/")
     public ResponseEntity<UserProfileStatisticsDto> getUserProfileStatistics(
-        @Parameter(description = "Id of current user. Cannot be empty.") @PathVariable @CurrentUserId Long userId) {
+            @PathVariable Long userId
+            , @Parameter(description = "Id of current user. Cannot be empty.") @CurrentUserId Long currentUserId) {
+        if (!currentUserId.equals(userId)) {
+            return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
+        }
```

Comment:
Why did you modify this method? You have already fixed it in the different pull request.
How is it going to help fix the bug from the 215 task?